opencti-platform/opencti-front/src/utils/hooks/useComputeLink.tsx

```diff
@@ -41,4 +43,38 @@ const useComputeLink = () => {
   return computeLink;
 };
 
-export default useComputeLink;
+type ComputeLinkFn = ReturnType<typeof useComputeLinkFn>;
+
+const ComputeLinkContext = createContext<ComputeLinkFn | null>(null);
+
+export const useComputeLink = (): ComputeLinkFn => {
+  const computeLink = useContext(ComputeLinkContext);
+  if (!computeLink) {
+    throw new Error('useSafeComputeLink must be used within ComputeLinkProvider');
+  }
+  return computeLink;
+};
+
+const PrivateComputeLinkProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
+  const computeLink = useComputeLinkFn();
+  return (
+    <ComputeLinkContext.Provider value={computeLink as ComputeLinkFn}>
+      {children}
+    </ComputeLinkContext.Provider>
+  );
+};
+
+export const ComputeLinkProvider = ({ children }: { children: ReactNode }) => {
+  const location = useLocation();
```

Comment:
useLocation triggers a rerender on every navigation, which could cause unnecessary rerendering.

Since we only instantiate ComputeLinkProvider in two places, we could simply pass a prop to indicate whether it's a public or private one.

Reply:
Handled in https://github.com/OpenCTI-Platform/opencti/pull/12914